lib/remote/apiuser.cpp

```diff
@@ -47,9 +47,18 @@ ApiUser::Ptr ApiUser::GetByAuthHeader(const String& auth_header)
 	 */
 	if (!user || password.IsEmpty())
 		return nullptr;
-	else if (user && !Utility::ComparePasswords(password, user->GetPassword()))
-		return nullptr;
-
+	else if (user) {
+        String hashedPassword = user->GetHashedPassword();
+        if (hashedPassword.IsEmpty()) {
+            if (Utility::ComparePasswords(password, user->GetPassword())) {
+                return nullptr;
+            }
+        } else {
+            if (strcmp(hashedPassword.CStr(), crypt(password.CStr(), hashedPassword.CStr())) == 0) {
```

Comment:
Please move this into a dedicated function, e.g. Utility::CompareHashedPasswords() which returns a boolean value.
The next step is to add unit tests for this.

Reply:
... once it works at all.